addOns/ascanrulesAlpha/src/test/java/org/zaproxy/zap/extension/ascanrulesAlpha/Log4ShellScanRuleUnitTest.java

```diff
@@ -0,0 +1,123 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+import static fi.iki.elonen.NanoHTTPD.newFixedLengthResponse;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import fi.iki.elonen.NanoHTTPD;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.httpclient.URI;
+import org.junit.jupiter.api.Test;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.extension.ExtensionLoader;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.zaproxy.addon.oast.ExtensionOast;
+import org.zaproxy.addon.oast.services.interactsh.InteractshService;
+import org.zaproxy.zap.extension.script.ExtensionScript;
+import org.zaproxy.zap.model.Tech;
+import org.zaproxy.zap.model.TechSet;
+import org.zaproxy.zap.network.HttpRequestBody;
+import org.zaproxy.zap.testutils.NanoServerHandler;
+
+class Log4ShellScanRuleUnitTest extends ActiveScannerTest<Log4ShellScanRule> {
+
+    private static final String OAST_PAYLOAD = "emFwcm9ja3MxMjM0NTY3ODkwCg.example.com";
+
+    @Override
+    protected Log4ShellScanRule createScanner() {
+        return new Log4ShellScanRule();
+    }
+
+    @Test
+    void shouldTargetJavaApps() {
+        // Given
+        TechSet techSet = techSet(Tech.JAVA, Tech.SPRING);
+        ;
+        // Then
+        assertThat(rule.targets(techSet), is(equalTo(true)));
+    }
+
+    @Test
+    void shouldSendRequestWithCorrectPayload() throws Exception {
+        // Given
+        startServer();
+        ExtensionLoader extensionLoader = mock(ExtensionLoader.class);
+        Model model = mock(Model.class);
+        Control.initSingletonForTesting(model, extensionLoader);
+        ExtensionOast extOast = mock(ExtensionOast.class);
+        when(extensionLoader.getExtension(ExtensionOast.class)).thenReturn(extOast);
+        when(extensionLoader.getExtension(ExtensionScript.class))
+                .thenReturn(mock(ExtensionScript.class));
```

Comment:
Is this one really needed?

Reply:
Yes, the test case errors out otherwise. It's required by VariantFactory#getExtension, which is a result of the call to VariantFactory#createVariants under AbstractAppParamPlugin#scan(). We could also return null instead of the mocked ExtensionScript object...

Reply:
To be more precise I'm getting "Strict stubbing argument mismatch" if I remove this.

Reply:
OK.